wandb/sdk/wandb_login.py

```diff
@@ -177,26 +177,27 @@ def update_session(
     def _prompt_api_key(self) -> Tuple[Optional[str], ApiKeyStatus]:
 
         api = Api(self._settings)
-        try:
-            key = apikey.prompt_api_key(
-                self._settings,
-                api=api,
-                no_offline=self._settings.force if self._settings else None,
-                no_create=self._settings.force if self._settings else None,
-            )
-        except ValueError as e:
-            # invalid key provided, try again
-            wandb.termerror(e.args[0])
-            return self._prompt_api_key()
-
-        except TimeoutError:
-            wandb.termlog("W&B disabled due to login timeout.")
-            return None, ApiKeyStatus.DISABLED
-        if key is False:
-            return None, ApiKeyStatus.NOTTY
-        if not key:
-            return None, ApiKeyStatus.OFFLINE
-        return key, ApiKeyStatus.VALID
+        while True:
+            try:
+                key = apikey.prompt_api_key(
+                    self._settings,
+                    api=api,
+                    no_offline=self._settings.force if self._settings else None,
+                    no_create=self._settings.force if self._settings else None,
+                )
+            except ValueError as e:
+                # invalid key provided, try again
+                wandb.termerror(e.args[0])
+                continue
+            except TimeoutError:
+                wandb.termlog("W&B disabled due to login timeout.")
+                return None, ApiKeyStatus.DISABLED
+            if key is False:
+                return None, ApiKeyStatus.NOTTY
+            if not key:
+                return None, ApiKeyStatus.OFFLINE
+            else:
+                return key, ApiKeyStatus.VALID
```

Comment:
i think this was better the way it was... else is strange here and not needed

Reply:
removed else